netutils/config/parser.py

```diff
@@ -603,3 +603,78 @@ class JunosConfigParser(BaseSpaceConfigParser):
 
     comment_chars = []
     banner_start = []
+
+
+class ASAConfigParser(CiscoConfigParser):
+    """Cisco ASA implementation of ConfigParser Class."""
+
+    comment_chars = ["!", ":"]
+
+    def __init__(self, config):
+        """Create ConfigParser Object.
+
+        Args:
+            config (str): The config text to parse.
+        """
+        self.unique_config_lines = set()
+        self.same_line_children = set()
+        super(ASAConfigParser, self).__init__(config)
+
+    def _update_config_lines(self, config_line):
+        """Add a ``ConfigLine`` object to ``self.config_lines``.
+
+        In addition to adding entries to config_lines, this also updates:
+          * self.same_line_children
+          * self.unique_config_lines
+
+        Args:
+            config_line (str): The current config line being evaluated.
+
+        Returns:
+            None
+        """
+        super(ASAConfigParser, self)._update_config_lines(config_line)
+        entry = self.config_lines[-1]
+        if entry in self.unique_config_lines:
+            self.same_line_children.add(entry)
+        self.unique_config_lines.add(entry)
+
+    def build_config_relationship(self):
+        r"""Parse text tree of config lines and their parents.
+
+        Example:
+            >>> config = '''
+            ... interface Management0/0
+            ...  management-only
+            ...  nameif Management
+            ...  security-level 100
+            ...  ip address 10.1.1.10 255.255.255.0'''
+            >>> config_tree = ASAConfigParser(str(config))
+            >>> config_tree.build_config_relationship() == \
+            ... [
+            ...     ConfigLine(config_line="interface Management0/0", parents=()),
+            ...     ConfigLine(config_line=" management-only", parents=("interface Management0/0",)),
+            ...     ConfigLine(config_line=" nameif Management", parents=("interface Management0/0",)),
+            ...     ConfigLine(config_line=" security-level 100", parents=("interface Management0/0",)),
+            ...     ConfigLine(config_line=" ip address 10.1.1.10 255.255.255.0", parents=("interface Management0/0",)),
+            ... ]
+            True
+        """
+        for line in self.generator_config:
```

Comment:
@jmcgill298 Can you give this a quick review? I get lost a little bit what this is doing, but it seems like there is a logical reason to overwrite this method. 

Please feel free to review entire PR as everything looks good to me and I am inclined to merge.

Reply:
It was overwritten because the ASA does the banner differently. Instead of needing to parse it like it's a single line with a delimiter, you can just parse each line individually like normal since each line always starts with the banner keyword, ie:

```
banner motd <BANNER TEXT>
banner exec <BANNER TEXT>
```

Reply:
Thanks!! That makes sense.